packages/commonwealth/client/scripts/views/pages/DirectoryPage/useDirectoryPageData.tsx

```diff
@@ -0,0 +1,75 @@
+import ChainInfo from 'models/ChainInfo';
+import { useCommonNavigate } from 'navigation/helpers';
+import React, { useMemo } from 'react';
+import { useFetchRelatedCommunitiesQuery } from 'state/api/communities';
+import { CWCommunityAvatar } from 'views/components/component_kit/cw_community_avatar';
+import { CWText } from 'views/components/component_kit/cw_text';
+
+export enum ViewType {
+  Rows = 'Rows',
+  Tiles = 'Tiles',
+}
+interface UseDirectoryPageDataProps {
+  chainNodeId: number;
+  debouncedSearchTerm: string;
+  selectedViewType: ViewType;
+}
+const useDirectoryPageData = ({
+  chainNodeId,
+  debouncedSearchTerm,
+  selectedViewType,
+}: UseDirectoryPageDataProps) => {
+  const navigate = useCommonNavigate();
+  const { data: relatedCommunities } = useFetchRelatedCommunitiesQuery({
```

Comment:
Shouldn't we `model` the `relatedCommunities` response, like we do for most other responses?

Reply:
not sure exactly what you mean here? it is modeled so I can get autosuggestion on fields. Or you mean something else?

![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/cbf0f49f-8924-462b-bd57-bb57de865b60)